fem/coefficient.cpp

```diff
@@ -313,6 +313,31 @@ void MatrixFunctionCoefficient::Eval(DenseMatrix &K, ElementTransformation &T,
    }
 }
 
+void MatrixFunctionCoefficient::EvalSymmetric(Vector &K,
+                                              ElementTransformation &T,
+                                              const IntegrationPoint &ip)
+{
+   MFEM_VERIFY(symmetric && height == width && height < 4 && SymmFunction,
```

Comment:
Instead of introducing a new method, can't you check inside `Eval` in which case you are?

Reply:
The problem is the type of `K`. In the symmetric case, this function sets the upper triangular entries of `K` in a `Vector`, whereas `Eval` sets a `DenseMatrix`. They could both use a `DenseMatrix` with a small amount of unused entries in the symmetric case, so there are (at least) two options with pros and cons.